File: RayTracer/MaterialObject.fs
module RayTracer.MaterialObject

open System.Numerics
open RayTracer.Color
open RayTracer.Geometry

type MaterialObject(origin: Vector3, rotation : Quaternion, triangles: Triangle array, colorFn: SpectralColor-> SpectralColor) =
    inherit TracerObject(origin, rotation, triangles)
    member this.colorTransformation : SpectralColor -> SpectralColor = colorFn

File: RayTracerCs/MaterialObject.cs
using System.Collections.Generic;
using System.Numerics;

namespace RayTracerCs
{
    public class MaterialObject : TracerObject
    {
        public MaterialObject(
            Vector3 origin,
            Quaternion rotation,
            IEnumerable<Triangle> triangles,
            SpectralColor color
        ) : base(origin, rotation, triangles)
        {
            Color = color;
        }

        public SpectralColor Color { get; }
    }
}